core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -1663,10 +1666,10 @@ public String getDisplayName() {
         return Messages.Hudson_DisplayName();
     }
 
-    public synchronized List<JDK> getJDKs() {
-        if(jdks==null)
-            jdks = new ArrayList<JDK>();
-        return jdks;
+    public List<JDK> getJDKs() {
+        synchronized (jdksLock) {
+            return Collections.unmodifiableList(jdks);
```

Comment:
This will break any existing code that modifies the list returned by `getJDKs()`, e.g. https://github.com/jenkinsci/jenkins/blob/jenkins-1.643/test/src/test/java/hudson/model/ProjectTest.java#L332

Why hasn't this been tested by the pull request builder? This is not ready for merge.